x-pack/plugins/security_solution/public/management/services/exceptions_list/exceptions_list_api_client.ts

```diff
@@ -31,7 +31,13 @@ export class ExceptionsListApiClient {
   constructor(
     private readonly http: HttpStart,
     public readonly listId: ListId,
-    private readonly listDefinition: CreateExceptionListSchema
+    private readonly listDefinition: CreateExceptionListSchema,
+    private readonly readTransform?: (item: ExceptionListItemSchema) => ExceptionListItemSchema,
+    private readonly writeTransform?: <
+      T extends CreateExceptionListItemSchema | UpdateExceptionListItemSchema
+    >(
+      item: T
+    ) => T
```

Comment:
I would like @dasansol92 comment here.
Ideally , we should not have read/write transforms in this base class because it should all be using `ExceptionListItemSchema` and so there should be no need for transformation of data. I understand from our discussion over slack that this was in response to the UI using just a `hash` field for the user to enter any of the supported hashes, which we then calculate the `field` name based on the value entered (ex. `process.hash.md5|sha1|sha256`). 
I'm just not sure if pushing that data value manipulation down to the API service is the right thing to do - to me, it feels move like a UI component prop type of mapping when setting/reading the value the user entered.

That being said, having a "middleware" callback for read/write is not a bad idea, so I'm 👍 this for now. We should really discuss if we want to continue to use those `TrustedApp**` types, since we have been trying to remove all of them in support of only `ExceptionListItemSchema` 

_(FYI: the mappers you moved around were originally in the server API for trusted apps, but moved to the UI when we deleted the TA specific API and started using the List Exceptions API. Our intent was to then have a second pass through to delete all of the types and adjust TA to be just like every other artifact)_

cc/ @ashokaditya since he also recently moved around some TA types to the `@kbn/**` library

Reply:
After thinking a bit more about it  and after reading the comment above, I'm agree with what @paul-tavares has written. 

As this is needed for UI maybe would be better to have it in an upper level. Also, we have plans to remove this types so we can just add this on that removal plan.

I would say let's move forward with this approach but let's take note about it in order to move/remove it when we get rid of the `TA` types.

PS: I like the way @joeypoon included this as a kind of hooks in this class. Let's keep this in mind for future pre/post validations, modifications, transformations, etc in frontend side.